Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -2402,7 +2400,7 @@ - (WKTouchEventsGestureRecognizer *)touchEventGestureRecognizer
 
 - (NSArray<WKDeferringGestureRecognizer *> *)_touchStartDeferringGestures
 {
-    WKDeferringGestureRecognizer *recognizers[3];
+    std::array<WKDeferringGestureRecognizer *, 3> recognizers;
```

Comment:
While it’s slightly more efficient to use `std::array`, it seems like it would be nicer to use something where the counting and conversion to `NSArray` is more built in rather than hand coded. For example we could use a `Vector` with inline capacity and add only a tiny bit of overhead; would not add any memory allocation. And would be more obviously safe, even if the magic number `3` was wrong.

Reply:
And then we’d just use `makeNSArray` instead of calling the `NSArray` class method directly with multiple arguments, relieving us of the need to check that the two arguments are consistent.

Reply:
Looks like I could also keep using a std::array and just do:
```
return createNSArray(std::span { recognizers }.first(count)).autorelease();
```

Reply:
Although now I feel like it may simply be better to construct a NSMutableArray right away instead of requiring an intermediary `std::array` / `Vector`

Reply:
I think this is nice and efficient:
```
static void appendRecognizerIfNonNull(RetainPtr<NSMutableArray>& array, const RetainPtr<WKDeferringGestureRecognizer>& recognizer)
{
    if (recognizer)
        [array addObject:recognizer.get()];
}

- (NSArray<WKDeferringGestureRecognizer *> *)_touchStartDeferringGestures
{
    RetainPtr recognizers = adoptNS([[NSMutableArray alloc] initWithCapacity:3]);
    appendRecognizerIfNonNull(recognizers, _touchStartDeferringGestureRecognizerForImmediatelyResettableGestures);
    appendRecognizerIfNonNull(recognizers, _touchStartDeferringGestureRecognizerForDelayedResettableGestures);
    appendRecognizerIfNonNull(recognizers, _touchStartDeferringGestureRecognizerForSyntheticTapGestures);
    return recognizers.autorelease();
}
```